src/chocolatey/infrastructure.app/configuration/ConfigurationOptions.cs

```diff
@@ -120,10 +125,33 @@ public static OptionSet OptionSet
 
             if (configuration.HelpRequested)
             {
+                if (configuration.ShowOnlineHelp)
+                {
+                    var targetAddress = "https://ch0.co/c/{0}".FormatWith(configuration.CommandName.ToLowerSafe());
+
+                    try
+                    {
+                        System.Diagnostics.Process.Start(new ProcessStartInfo(targetAddress) { UseShellExecute = true });
+                    }
+                    catch (Exception)
+                    {
+                        "chocolatey".Log().Warn("There was an error while attempting to open the following URL: {0} in the default browser.".FormatWith(targetAddress));
+                    }
+
+                    return;
+                }
+
                 ShowHelp(_optionSet, helpMessage);
             }
             else
             {
+                // Only show this warning once
+                if (!_helpWarningHasBeenShown && configuration.ShowOnlineHelp)
+                {
+                    "chocolatey".Log().Warn("The --online option has been used, without the corresponding -?/--help/-h option.  Command execution will be completed without invoking help.");
+                    _helpWarningHasBeenShown = true;
+                }

```

Comment:
Something about this check isn't working as expected. It's showing for every invocation, even when I _do_ have the `--help` or `-h` option provided. 

Does this code path get hit multiple times for some reason, and one of the times it doesn't have `-h` set at that point?

Reply:
I have taken a stab at fixing this, can you please review this again? Thanks!